services/checks/checks.go

```diff
@@ -0,0 +1,185 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package checks provides security checks functionality.
+package checks
+
+import (
+	"context"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	api "github.com/percona-platform/saas/gen/checked"
+	"github.com/percona-platform/saas/pkg/check"
+	"github.com/percona/pmm/utils/tlsconfig"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	defaultHost     = "check.percona.com"
+	defaultInterval = 24 * time.Hour
+
+	// Environment variables that affect checks service; only for testing.
+	envHost      = "PERCONA_TEST_CHECKS_HOST"
+	envPublicKey = "PERCONA_TEST_CHECKS_PUBLIC_KEY"
+	envInterval  = "PERCONA_TEST_CHECKS_INTERVAL"
```

Comment:
We should add support for a static file with `PERCONA_TEST_CHECKS_FILE`. No signatures checking for this file, not hot reloading, etc. That's only for our own testing.